js/search/domain.js

```diff
@@ -22,7 +20,6 @@ const LINK_TABLE_ID = '4dba9091495e8f277893e0d400e9e092f9f6f551';
  */
 const Domain = new Lang.Class({
     Name: 'Domain',
-    Abstract: true,
```

Comment:
Agreed with @mattdangerw's comments above -- the port to C would have rather large changes for implementation, and I see no reason to carry that over -- we can reintroduce a V4 Domain with an abstract class implementation later.

I'm not convinced that EKN_VERSION should be tied to runtime version -- I can see it argued both ways. We might want to upgrade e.g. Prensa Libre to a new version for a new module without having to stop old versions of the app from working, doing large content migration, or changing subscriptions. But that's probably for another time.